src/NHibernate/Linq/ExpressionTransformers/LikeTransformer.cs

```diff
@@ -0,0 +1,60 @@
+using System.Linq.Expressions;
+using System.Reflection;
+using NHibernate.Util;
+using Remotion.Linq.Parsing.ExpressionVisitors.Transformation;
+
+namespace NHibernate.Linq.ExpressionTransformers
+{
+	/// <summary>
+	/// Replace <see cref="string.StartsWith(string)"/>, <see cref="string.EndsWith(string)"/> and <see cref="string.Contains(string)"/>
+	/// with <see cref="SqlMethods.Like(string, string)"/>
+	/// </summary>
+	internal class LikeTransformer : IExpressionTransformer<MethodCallExpression>
+	{
+		private static readonly MethodInfo Like = ReflectHelper.FastGetMethod(SqlMethods.Like, default(string), default(string));
+
+		public ExpressionType[] SupportedExpressionTypes { get; } = {ExpressionType.Call};
+
+		public Expression Transform(MethodCallExpression expression)
+		{
+			if (expression.Method.Name == nameof(string.StartsWith) && expression.Method == ReflectionCache.StringMethods.StartsWith)
```

Comment:
Why do you think it optimizes anything?

Reply:
What exactly do you mean?

Reply:
I just don't understand this optimization with method name comparison. Are you aware of some  heavy internal method equals logic?

Reply:
Quick fiddle test shows no performance gain. It's only lost:
https://dotnetfiddle.net/Di9PUY 

Did I miss something?

Reply:
Ok, reverted.